packages/web3-eth-contract/src/contract.ts

```diff
@@ -132,17 +172,132 @@ export class Contract<Abi extends ContractAbi>
 		this._jsonInterface = [...result] as unknown as Abi;
 	}
 
-	// eslint-disable-next-line class-methods-use-this
-	private _createContractFunction(
-		name: string,
-		// eslint-disable-next-line @typescript-eslint/no-unused-vars
-		_abi: AbiFunctionFragment,
-	): ContractBoundFunction {
-		return name;
+	private _createContractMethod(abi: AbiFunctionFragment): ContractBoundMethod {
+		return (...params) => {
+			validator.validate(abi.inputs, params);
+
+			if (abi.stateMutability === 'payable' || abi.stateMutability === 'pure') {
+				return {
+					arguments: params,
+					call: async (options?: PayableCallOptions, block?: BlockNumberOrTag) =>
+						this._contractMethodCall(abi, params, options, block),
+					// TODO: Use `web3-eth-tx` package to return `PromiEvent` instead.
```

Comment:
Perhaps we should be creating [web3-core-promievent](https://github.com/ChainSafe/web3.js/tree/1.x/packages/web3-core-promievent)

Reply:
The `PromiEvent` is already created in the `web3-common`.